CONTRIBUTING.md

```diff
@@ -1,54 +1,57 @@
 # CONTRIBUTING
-
-Contributing to the official buyTogether Application
+Contributing to the official buyTogether application
 Welcome! We're so excited that you're considering contributing to the official GitHub page for buyTogether.
-
 ---
-
-## Make changes locally
-
-- Fork the repo so that you can make your changes without affecting the original project until you're ready to merge them.
-
-- Create a working branch 
-
-- Set up a development environment to begin working on the project all you need to do is 'vagrant up'
-
+## Prerequisites
+- Fork the repo to make your changes without affecting the original project until you're ready to merge.
+- Create a working branch.
+- Set up a development environment to begin working on the project, all you need to do is 'vagrant up'
 - Start with your changes!
-
 ---
-
-## Commit your update
-Commit the changes once you are happy with them. 
+## Commiting
+Please use the guide below inorder to follow our commit message convention:
+* [commit convention](https://www.conventionalcommits.org/en/v1.0.0/)
 
 ---
-
 ## Pull Request
 - When you're finished with the changes, create a pull request, also known as a PR.
-- Please note that at least two team members of the group should review and approve the pull request before asking the mentors to review and merge.
-- The reviewer resolve the conversation to indicate that they are ok with the updated content.
+- If your PR depends on the merging of other PRs, indicate this on the commit message or on the comments.
+Specifically, if it contains unmerged commits of any other PRs,
+indicate to the reviewers which commits are expected to be reviewed as part of your PR.
+- Keep your PRs as small as possible.
+Ideally, a single PR should include a single commit.
+- The reviewer resolves the conversation to indicate that they are ok with the updated content.
 - If you run into any merge issues, check how to solve them and resolve merge conflicts and other issues.
-
+- Changes made as part of the review process (i.e. fixes to the content of the original commit) ,
+should be added to the original commit (e.g. using ```commit --amend```), rather than creating a new commit. 
+- Please keep in mind that at least two team members of the group should "quality review"(i.e. NO LGTM) and approve the pull request before asking the mentors to review and merge it.
+- Each PR should include testing(Code testing section below).
+PR without tests won't be reviewed.
 ---
+## Code Testing
+Please use Pytest as your test framework.
+
+* [Pytest](https://docs.pytest.org/en/7.2.x/getting-started.html)
+- Please note that Pytest is already installed in your virtual environment.
 
+---
+## Review requirements 
+- When reviewing, confirm that the code accomplishes the requirements and the acceptance criteria,
```

Comment:
Need to define, the requirements and acceptance criteria of what. GitHub issue? commit message? Something else?

If it's a GitHub issue, then you should define that each PR should be [tied to an issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue). If it's the commit message, then you need more elaborated criteria for the commit message content.

I think linking PRs to issues is easier to accomplish and more flexible.

Reply:
i think i will just delete this section, i dont find it much informative and i dont know how to elaborate it properly.

Reply:
but that is how you decide whether the PR is doing what it's suppose to be doing or not. If you don't have that, then you can't tell whether the PR is actually doing what it's supposed to be doing.

Reply:
You can say here that the PR needs to either be linked to a GitHub issue, or should specify in details what it's supposed to be achieving.

Reply:
added your notes, thanks!